src/System.IO/src/System/IO/BinaryWriter.cs

```diff
@@ -32,7 +33,7 @@ public class BinaryWriter : IDisposable
         protected BinaryWriter()
         {
             OutStream = Stream.Null;
-            _buffer = new byte[16];
+            _buffer = ArrayPool<byte>.Shared.Rent(16);
```

Comment:
There would be a lot of duplicate code since lots of different functions use _buffer, so I thought it made it simpler to initialize it in the ctor, especially since it is a trivially small array.